src/beignet/splits/_murcko_scaffold_split.py

```diff
@@ -0,0 +1,110 @@
+import math
+import random
+from collections import defaultdict
+
+from torch.utils.data import Dataset, Subset
+
+try:
+    from rdkit import Chem
+    from rdkit.Chem.Scaffolds.MurckoScaffold import MurckoScaffoldSmiles
+
+    _RDKit_AVAILABLE = True
+except (ImportError, ModuleNotFoundError):
+    _RDKit_AVAILABLE = False
+    Chem, MurckoScaffoldSmiles = None, None
+
+
+def murcko_scaffold_split(
+    dataset: Dataset,
+    smiles: list[str],
+    test_size: float | int,
+    *,
+    seed: int = 0xDEADBEEF,
+    shuffle: bool = True,
+    include_chirality: bool = False,
+) -> tuple[Subset, Subset]:
+    """Split a dataset based on Murcko scaffold splitting based
```

Comment:
Prepend `r`

Reply:
Can you elaborate? Is that for the input SMILES strings? Escape characters shouldn't be present in valid SMILES anyway and isn't it that if a string is formed with escape chars without r initially, prepending it won't change the format? Or maybe I'm just misunderstanding

Reply:
No, preprend `r` to the docstring.